pandas/core/reshape/reshape.py

```diff
@@ -1094,6 +1097,208 @@ def get_empty_frame(data) -> DataFrame:
+    data : `DataFrame`
+        Data which contains dummy-coded variables.
+    subset : None, Index, or list of Hashables, default 'None'
+        The columns which to convert from dummy-encoding and return as categorical
+        `DataFrame`. If `columns` is None then all dummy columns are converted and
+        appended to the non-dummy columns.
+    sep : str or dict of str, default '_'
+        Separator used in the column names of the dummy categories they are
+        character indicating the separation of the categorical names from the prefixes.
+        For example, if your column names are 'prefix_A' and 'prefix_B',
+        you can strip the underscore by specifying sep='_'.
+        Alternatively, pass a dictionary to map prefix separators to prefixes if
+        multiple and/or mixed separators are used in the column names.
+    dropped_fist : None, Hashable or dict of Hashables, default None
+        The implied value the dummy takes when all values are zero.
+        Can be a a single value for all variables or a dict directly mapping the
+        dropped value to a prefix of a variable.
+
+    Returns
+    -------
+    `DataFrame`
+        Categorical data decoded from the dummy input-data.
+
+    See Also
+    --------
+    get_dummies : Convert `Series` or `DataFrame` to dummy codes.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0],
+    ...                   "c": [0, 0, 1, 0]})
+
+    >>> pd.from_dummies(df)
+    0     a
+    1     b
+    2     c
+    3     a
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 1], "col1_b": [0, 1, 0],
+    ...                   "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                   "col2_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep="_")
+        col1    col2
+    0    a       b
+    1    b       a
+    2    a       c
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 0], "col1_b": [0, 1, 0],
+    ...                    "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                    "col2_c": [0, 0, 0]})
+
+    >>> pd.from_dummies(df, sep="_", dropped_first={"col1": "d", "col2": "e"})
+        col1    col2
+    0    a       b
+    1    b       a
+    2    d       e
+
+    >>> df = pd.DataFrame({"col1_a-a": [1, 0, 1], "col1_b-b": [0, 1, 0],
+    ...                   "col2-a_a": [0, 1, 0], "col2-b_b": [1, 0, 0],
+    ...                   "col2-c_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep={"col1": "_", "col2": "-"})
+       col1  col2
+    0  a-a   b_b
+    1  b-b   a_a
+    2  a-a   c_c
+    """
+    from pandas.core.reshape.concat import concat
+
+    if subset is None:
+        subset = list(data.columns)
+    elif isinstance(subset, Index):
+        subset = list(subset)
+    elif not isinstance(subset, list):
+        raise TypeError(
+            f"Expected 'subset' to be of type 'Index', or 'list'; "
+            f"Received 'subset' of type: {type(subset).__name__}"
+        )
+
+    if data[subset].isna().any().any():
```

Comment:
if this is required then put this in a Notes section to inform the user.

Reply:
> please make this much simpler. you are adding a ton of special case implementation. Let's do the bare bones implementation and nothing else.

It should be much simpler now, I removed many cases that can be done by before using the function. I realized that e.g. it does not make a huge difference to pass the subset of the DataFrame that is supposed to be converted instead of passing the entire DataFrame and a subset argument but it makes the funciton cleaner and simpler (Same line of thougt for different types of separators).

The only thing I did not strip down is the `implied_value` argument as I think this is quite important for the conversion.